### PR TITLE
Bug fixes for gradient calculation

### PR DESCRIPTION
**Pull request by rishit2307**
Gradient computation now done by all ranks

### CODE REVIEW

File: pyfr/plugins/tavg.py [hunk 1]
```diff
@@ -117,8 +121,8 @@ def _eval_acc_exprs(self, intg):
 
             # Prepare any required gradients
             if self._gradpinfo:
-                # Compute the gradients
-                grad_soln = np.rollaxis(intg.grad_soln[idx], 2)[..., rgn]
+                # Fetch the gradients
```
Comment (FreddieWitherden) [inline pyfr/plugins/tavg.py:124]: Subset

File: pyfr/plugins/tavg.py [hunk 2]
```diff
@@ -105,6 +105,10 @@ def _eval_acc_exprs(self, intg):
         # Get the primitive variable names
         pnames = self.elementscls.privarmap[self.ndims]
 
+        #Compute gradients
```
Comment (FreddieWitherden) [inline pyfr/plugins/tavg.py:108]: Space after #

File: pyfr/plugins/tavg.py (hunk 1 above)
Comment (rishit2307) [reply pyfr/plugins/tavg.py:124]: You mean change the comment to # Fetch subset of gradients? Or something more appropriate?

**Comment by FreddieWitherden** [comment]
Rebase so it is just a single commit on top of develop.

**Pull request merged**